src/lib/storage/table.cpp

```diff
@@ -144,7 +151,14 @@ void Table::remove_chunk(ChunkID chunk_id) {
   _chunks[chunk_id] = nullptr;
 }
 
-void Table::append_chunk(const Segments& segments, const std::optional<PolymorphicAllocator<Chunk>>& alloc) {
+void Table::append_chunk(const Segments& segments, std::shared_ptr<MvccData> mvcc_data,
+                         const std::optional<PolymorphicAllocator<Chunk>>& alloc) {
+  DebugAssert(_type != TableType::References || !mvcc_data,
+              "Setting explicit MvccData on a reference Table makes no sense. Reference Tables should use MvccData of "
+              "referenced Table, if any");
+  DebugAssert(_type != TableType::Data || static_cast<bool>(mvcc_data) == (_use_mvcc == UseMvcc::Yes),
+              "Supply MvccData to data Tables iff MVCC is enabled");
```

Comment:
```suggestion
              "Supply MvccData to data Tables if MVCC is enabled");
```

Reply:
[That](https://en.wikipedia.org/wiki/If_and_only_if)'s actually intended:
Data Table with MVCC enabled -> You **must** supply MvccData
Data Table without MVCC enabed -> You **cannot** supply MvccData

Reply:
Sounds good. Always learning something new. ;)